### ISSUE TITLE
Where do I find any instruction on how do use  cri-o with kubernetes  ?

### ISSUE THREAD

**Issue by whatdhack** [kubernetes-incubator/cri-o#344]
I could not find any instruction on how to run kubernetes with cri-o .  Would appreciate any pointer.

**Comment by runcom** [comment]
Right now, I'm testing kubernetes and CRI-O with the following:
```
EXPERIMENTAL_CRI=true CONTAINER_RUNTIME=remote CONTAINER_RUNTIME_ENDPOINT='/var/run/ocid.sock --runtime-request-timeout=15m' ./hack/local-up-cluster.sh
```
That's basically running kubernetes locally. Note, you have to start `ocid` before bring up kubernetes!

I guess we need to add this information to the README.

**Comment by rhatdan** [comment]
Should we add a unit file to ocid to configure kubernetes to run with ocid.  Take kubernetes systemd unit file in Fedora 26 and modify it to run with ocid rather then docker.  


Can we override it by dropping a file in /etc/systemd/system/kubernetes.d/ocid.service.

**Comment by cyphar** [comment]
Maybe, but not all distributions have the same Kubernetes setup. I'd prefer to have some solid docs in `README` which say exactly what flags and switches you need for the `kubelet` in order for it to use `ocid` (as well as whatever other parts of Kubernetes need to be modified).

**Comment by feiskyer** [comment]
> EXPERIMENTAL_CRI=true CONTAINER_RUNTIME=remote CONTAINER_RUNTIME_ENDPOINT='/var/run/ocid.sock --runtime-request-timeout=15m' ./hack/local-up-cluster.sh

@runcom It's better to include this in README.

**Comment by Bregor** [comment]
Is there possibility to do it without `kube-up`?
Args for `kubelet`, etc?

**Comment by cyphar** [comment]
@Bregor Yes, but it's a bit of a pain at the moment with the `kubelet` switches which is why I said above that we need to also document the `kubelet` switches necessary (so you don't have to do kube-up).


From a quick look at the man page this is what appears to be necessary:
```
kubelet --experimental-cri --container-runtime-endpoint="/var/run/ocid.sock" --runtime-request-timeout="15m"
```

Though you might need to also set `--container-runtime` to a different value (though the man page appears to indicate that you can only use `rkt` and `docker` (neither of which is `ocid`). Maybe it should be `--container-runtime=remote`?

**Comment by runcom** [comment]
> Though you might need to also set --container-runtime to a different value (though the man page appears to indicate that you can only use rkt and docker (neither of which is ocid). Maybe it should be --container-runtime=remote?

yes, it's `--container-runtime=remote` - you can follow what `./hack/local-up-cluster.sh` is doing when preparing `kubelet` args using my original comment https://github.com/kubernetes-incubator/cri-o/issues/344#issuecomment-275788358

**Comment by Bregor** [comment]
@cyphar, @runcom thank you, I will try

**Comment by jjlakis** [comment]
I am running cri-o on the k8s **cluster**, but I started with docker cluster and switched runtime to cri-o on each node. If you would like to include it as part of README, I can do it.

**Comment by runcom** [comment]
> I am running cri-o on the k8s cluster, but I started with docker cluster and switched runtime to cri-o on each node. If you would like to include it as part of README, I can do it.

thanks @jjlakis feel free to make a PR if you want!

**Comment by runcom** [comment]
fixed by #353
